homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -57,17 +61,23 @@
     ip_address = config[CONF_IP_ADDRESS]
     monitored_conditions = config[CONF_MONITORED_CONDITIONS]
     name = config[CONF_NAME]
+    username = config[CONF_USERNAME]
+    password = config[CONF_PASSWORD]
 
     entities = []
     # Iterate through the list of sensors
     for condition in monitored_conditions:
         if condition == "inverters":
-            inverters = await EnvoyReader(ip_address).inverters_production()
+            inverters = await EnvoyReader(
+                ip_address, username, password
+            ).inverters_production()
```

Comment:
I assume that there will be a traceback if the credentials are wrong. If that's the case then the platform setup must fail otherwise the users will end-up with a non-functional platform in their setup.

Also, the `EnvoyReader()` should be passed to the `Envoy` class and then used by `async_update()`.

Reply:
@fabaff The Envoy has both unsecured and secured HTML pages.  Currently the API uses a default username and a password that is derived from the Envoy's serial number, which is used for the secured pages.  Even if the API can not log in to the secured pages, the unsecure pages are still accessible.  So with the released sensor in HA, the platform is functional just with minimal data.

I was just adding a way for the user to be able to change the default username and password, so that the API can return the secured page info from their Envoy device (like the one I have) to the HA sensor.

I understand your concern but the current sensor and API, credentials are failing without affecting the functionality of the platform.  Now with that said logging on the API side can be better to highlight that the credentials have failed.  That I can look into.

Reply:
I updated the sensor so that the `EnvoyReader()` is passed to the `Envoy` class.

Reply:
Is the user informed now somehow if credentials are bad?

Reply:
Currently no.  I can look into getting some message into the HA log, but it might require a change on the API side.

Reply:
I added a Warning message when the Authentication fails for the Inverter data.  I thought about putting it at the Error level, but the sensor still functions when authentication fails just without any Inverter data.  For the message to appear in the HA log, a code change is required on the API side, here is that PR [#18](https://github.com/jesserizzo/envoy_reader/pull/18)

```
except requests.exceptions.HTTPError:
    _LOGGER.warning(
        "Authentication for Inverter data failed during setup: %s",
        ip_address,
    )
```